File: completed/2264.java
class Solution {
    public String largestGoodInteger(String num) {
        int[] f = new int[num.length()];
        int largest = -1;
        for (int i = 1; i < num.length(); i++) {
            if (num.charAt(i) == num.charAt(i - 1)) {
                f[i] = f[i - 1] + 1;
                if (f[i] >= 2) {
                    largest = Math.max(Character.getNumericValue(num.charAt(i)), largest);
                }
            }
        }
        return largest == -1 ? "" : (largest + "").repeat(3);
    }

    public static void main(String[] args) {
        String num = "6777133339";

        Solution solution = new Solution();
        System.out.println(solution.largestGoodInteger(num));
    }
}

File: completed/2264.cpp
class Solution {
public:
    string largestGoodInteger(string num) {
        int largest = -1;
        int left = 0;
        while (left < num.length()) {
            int right = left + 1;
            while (right < num.length() && num[right] == num[left]) {
                right++;
            }

            if (right - left >= 3) {
                largest = max((int) num[left] - '0', largest);
            }
            left = right;
        }

        if (largest == -1) {
            return "";
        }

        return string(3, (char) largest + '0');
    }
};
